Few issues with static analyzer

I'm looking at the static analyzer report https://cmssdt.cern.ch/SDT/jenkins-artifacts/ib-static-analysis/CMSSW_13_0_X_2022-12-11-2300/el8_amd64_gcc11/llvm-analysis/index.html with only the "fast math plugin" checks enabled, and I see
![image](https://user-images.githubusercontent.com/33993/207074567-91e445e7-6df8-48d9-89fc-006af7dcd086.png)

which makes me wonder a couple of things
1. How does `DataFormats/PatCandidates/src/PackedCandidate.cc` get flagged? As far as I can tell, [`DataFormats/PatCandidates/BuildFile.xml`](https://github.com/cms-sw/cmssw/blob/master/DataFormats/PatCandidates/BuildFile.xml) does not declare fast math to be used
2. I see five reports with numbers in "Bug Group" and "Bug Type". Clicking to the "View Report" link shows that all are `edm::ParameterSet::existsAs<>()`. Could there be something that causes these not to be collected under "Function edm::ParameterSet::exists() or edm::ParameterSet::existsAs<>() called" type?
3. I checked a few of the first analyzer failure reports, and they pointed to an assertion failure (caused by `nullptr`) in `clangcms::ConstCastAwayChecker::checkPreStmt()` along
```
clang++: /cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02763/el8_amd64_gcc11/external/llvm/12.0.1-476112d9475f69ecba350f77e3ec4975/include/llvm/Support/Casting.h:104: static bool llvm::isa_impl_cl<To, const From*>::doit(const From*) [with To = clang::AttributedStmt; From = clang::Stmt]: Assertion `Val && "isa<> used on a null pointer"' failed.

...

1.	<eof> parser at end of file
2.	While analyzing stack: 
	#0 Calling HcalQIESample::HcalQIESample at line 19
	#1 Calling CastorNominalCoderTemplate::process at line 30
	#2 Calling CastorNominalCoder::fC2adc
3.	/pool/condor/dir_35136/jenkins/workspace/ib-run-static-checks/CMSSW_13_0_X_2022-12-11-2300/src/DataFormats/HcalDigi/interface/HcalQIESample.h:37:19: Error evaluating statement
4.	/pool/condor/dir_35136/jenkins/workspace/ib-run-static-checks/CMSSW_13_0_X_2022-12-11-2300/src/DataFormats/HcalDigi/interface/HcalQIESample.h:37:19: Error evaluating statement
 #0 0x00002b12fd51975e llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02763/el8_amd64_gcc11/external/llvm/12.0.1-476112d9475f69ecba350f77e3ec4975/bin/../lib64/libLLVM-12.so+0xa0175e)
 #1 0x00002b12fd5174f4 llvm::sys::RunSignalHandlers() (/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02763/el8_amd64_gcc11/external/llvm/12.0.1-476112d9475f69ecba350f77e3ec4975/bin/../lib64/libLLVM-12.so+0x9ff4f4)
 #2 0x00002b12fd51765d SignalHandler(int) (/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02763/el8_amd64_gcc11/external/llvm/12.0.1-476112d9475f69ecba350f77e3ec4975/bin/../lib64/libLLVM-12.so+0x9ff65d)
 #3 0x00002b12f931bcf0 __restore_rt (/lib64/libpthread.so.0+0x12cf0)
 #4 0x00002b1301371aff raise (/lib64/libc.so.6+0x4eaff)
 #5 0x00002b1301344ea5 abort (/lib64/libc.so.6+0x21ea5)
 #6 0x00002b1301344d79 _nl_load_domain.cold.0 (/lib64/libc.so.6+0x21d79)
 #7 0x00002b130136a456 (/lib64/libc.so.6+0x47456)
 #8 0x00002b12f91c0006 clangcms::ConstCastAwayChecker::checkPreStmt(clang::ExplicitCastExpr const*, clang::ento::CheckerContext&) const (/pool/condor/dir_35136/jenkins/workspace/ib-run-static-checks/CMSSW_13_0_X_2022-12-11-2300/lib/el8_amd64_gcc11/libUtilitiesStaticAnalyzers.so+0x3b006)
 #9 0x00002b12fbc1958d clang::ento::CheckerManager::runCheckersForStmt(bool, clang::ento::ExplodedNodeSet&, clang::ento::ExplodedNodeSet const&, clang::Stmt const*, clang::ento::ExprEngine&, bool) (/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02763/el8_amd64_gcc11/external/llvm/12.0.1-476112d9475f69ecba350f77e3ec4975/bin/../lib64/libclang-cpp.so.12+0x26f058d)
```

--- a/Utilities/StaticAnalyzers/src/ConstCastAwayChecker.cpp
+++ b/Utilities/StaticAnalyzers/src/ConstCastAwayChecker.cpp
@@ -24,16 +24,16 @@ namespace clangcms {
     if (!(clang::CStyleCastExpr::classof(CE) || clang::CXXConstCastExpr::classof(CE)))
       return;
     auto P = C.getCurrentAnalysisDeclContext()->getParentMap().getParent(CE);
-    while (!(isa<AttributedStmt>(P) || isa<DeclStmt>(P)) &&
+    while (P && !(isa<AttributedStmt>(P) || isa<DeclStmt>(P)) &&
            C.getCurrentAnalysisDeclContext()->getParentMap().hasParent(P)) {
       P = C.getCurrentAnalysisDeclContext()->getParentMap().getParent(P);
     }
-    if (isa<AttributedStmt>(P)) {
+    if (P && isa<AttributedStmt>(P)) {
       const AttributedStmt *AS = dyn_cast_or_null<AttributedStmt>(P);
       if (AS && (hasSpecificAttr<CMSSaAllowAttr>(AS->getAttrs()) || hasSpecificAttr<CMSThreadSafeAttr>(AS->getAttrs())))
         return;
     }
-    if (isa<DeclStmt>(P)) {
+    if (P && isa<DeclStmt>(P)) {
       const DeclStmt *DS = dyn_cast_or_null<DeclStmt>(P);
       if (DS && (hasSpecificAttr<CMSSaAllowAttr>(DS->getSingleDecl()->getAttrs()) ||
                  hasSpecificAttr<CMSThreadSafeAttr>(DS->getSingleDecl()->getAttrs())))

--- a/Utilities/StaticAnalyzers/src/ConstCastChecker.cpp
+++ b/Utilities/StaticAnalyzers/src/ConstCastChecker.cpp
@@ -21,16 +21,16 @@ namespace clangcms {
 
   void ConstCastChecker::checkPreStmt(const clang::CXXConstCastExpr *CE, clang::ento::CheckerContext &C) const {
     auto P = C.getCurrentAnalysisDeclContext()->getParentMap().getParent(CE);
-    while (!(isa<AttributedStmt>(P) || isa<DeclStmt>(P)) &&
+    while (P && !(isa<AttributedStmt>(P) || isa<DeclStmt>(P)) &&
            C.getCurrentAnalysisDeclContext()->getParentMap().hasParent(P)) {
       P = C.getCurrentAnalysisDeclContext()->getParentMap().getParent(P);
     }
-    if (isa<AttributedStmt>(P)) {
+    if (P && isa<AttributedStmt>(P)) {
       const AttributedStmt *AS = dyn_cast_or_null<AttributedStmt>(P);
       if (AS && (hasSpecificAttr<CMSSaAllowAttr>(AS->getAttrs()) || hasSpecificAttr<CMSThreadSafeAttr>(AS->getAttrs())))
         return;
     }
-    if (isa<DeclStmt>(P)) {
+    if (P && isa<DeclStmt>(P)) {
       const DeclStmt *DS = dyn_cast_or_null<DeclStmt>(P);
       if (DS && (hasSpecificAttr<CMSSaAllowAttr>(DS->getSingleDecl()->getAttrs()) ||
                  hasSpecificAttr<CMSThreadSafeAttr>(DS->getSingleDecl()->getAttrs())))

